moveit_core/robot_state/include/moveit/robot_state/robot_state.h

```diff
@@ -85,8 +85,31 @@ struct JumpThreshold
 
   explicit JumpThreshold(double jt_revolute, double jt_prismatic) : JumpThreshold()
   {
-    revolute = jt_revolute;
     prismatic = jt_prismatic;
+    revolute = jt_revolute;
+  }
+};
+
+/** \brief Struct for containing max_step for computeCartesianPath
+
+    Setting translation to zero will disable checking for translations and the same goes for rotation */
+struct MaxEEFStep
+{
+  double translation;
+  double rotation;
+
+  explicit MaxEEFStep() : translation(0.0), rotation(0.0)
+  {
+  }
+
+  explicit MaxEEFStep(double max_translation) : MaxEEFStep()
+  {
+    translation = max_translation;
+  }
+
+  explicit MaxEEFStep(double max_translation, double max_rotation)
+    : translation(max_translation), rotation(max_rotation)
+  {
```

Comment:
i don't think you need all these constructor types. checkout aggregate initialization in C++11
http://en.cppreference.com/w/cpp/language/aggregate_initialization